File: Javascript/SelectionSort.js
const arr = [8, 5, 7, 3, 2];

function Selection_Sort(A) {
  //length of array
  const n = arr.length;
  // for pass
  for (let i = 0; i < n - 1; i++) {
    //for comparison and swaping
    let j = i;
    let k = i;
    for (; j < n; j++) {
      if (A[j] < A[k]) {
        k = j;
      }
    }
    //swap
    let temp = A[i];
    A[i] = A[k];
    A[k] = temp;
  }
  //displaying after sort
  for (let i = 0; i < n; i++) console.log(A[i]);
}

Selection_Sort(arr);

// complexity = O(n^2)
// stable = False
// Adaptive

File: C++/selectionSort.cpp
#include<iostream>
#include<conio.h>

using namespace std;



void swap(int *a, int *b){
	int temp =  *a;
	*a = *b;
	*b = temp;
}



void selection_sort(int arr[], int n){
	for(int i=0;i<n-1;++i){
		int k = i;
		for(int j=i;j<n;j++){
			if(arr[j]<arr[k]){
				k = j;
			}
		}
		swap(arr[i], arr[k]);
	}
}

void display(int arr[], int n){
	for(int i=0;i<n;++i){
		cout<<arr[i]<<" ";
	}
	cout<<endl;
}


int main(){
	int arr[] = {8,6,3,2,5,4};
	int n = 6;
	
	//before sort
	display(arr, n);
	
	//sort
	selection_sort(arr, n);
	
	//after display
	display(arr, n);
	
}
